src/components/schedule-page/new-month-page.component.tsx

```diff
@@ -11,30 +11,47 @@ import { Button } from "../common-components";
 import { useScheduleConverter } from "./import-buttons/hooks/use-schedule-converter";
 import { ScheduleKey } from "../../api/persistance-store.model";
 import { MonthSwitchActionCreator } from "../../state/reducers/month-state/schedule-data/month-switch.action-creator";
+import { LocalStorageProvider } from "../../api/local-storage-provider.model";
+import { MonthDataModel } from "../../common-models/schedule-data.model";
 
 export function NewMonthPlanComponent(): JSX.Element {
   const state = useSelector((state: ApplicationStateModel) => state);
   const currentMonth = state.actualState.persistentSchedule.present.schedule_info.month_number;
   const currentYear = state.actualState.persistentSchedule.present.schedule_info.year;
+  const { revision } = state.actualState;
```

Comment:
Generalnie nie jest najleszą praktyką użycie jednego `useSelector`, który wyciąga globalny obiekt, bo wtedy jest większa szansa na to, że kompnent będzie renderować się od nowa ( przy każdym updacie  któregoś z pół naszego stanu )

Chociaż nie wiem czy w naszym przypadku to stanowi większy problem, ponieważ z tego co pamiętam, to my zawsze updatujemy cały stan. 
Oprócz momentu w którym my dodajemy błędy do grafiku.
Więc w tym przypadku to by renderowało się od nowa, ale dlatego komponentu on nigdy nie zajdzie, ponieważ nie mamy możliwości kopiowania sprawdzenia grafiku będąc na nim

Wiem, że kod nie z tego PR'a, ale rzuciło mi się w oczy, więc nie mogłem się powstrzymać :D

https://redux.js.org/style-guide/style-guide#call-useselector-multiple-times-in-function-components

Reply:
Dobre, czujne oko :D Optymalizacja, ale mała rzecz, więc ją wprowadzę :)